pubsub/subscription.go

```diff
@@ -925,13 +925,29 @@ func (s *Subscription) Receive(ctx context.Context, f func(context.Context, *Mes
 						}
 					}
 				}
+				// If the context is done, don't pull more messages.
+				select {
+				case <-ctx.Done():
+					return nil
```

Comment:
Should we be returning the ctx.Err() in these cases?

Reply:
I think in general we don't want `Receive` to return any errors, including context cancellation errors.

Reply:
Ack